internal/boxcli/global.go

```diff
@@ -10,73 +10,35 @@ import (
 	"github.com/spf13/cobra"
 
 	"go.jetpack.io/devbox"
-	"go.jetpack.io/devbox/internal/boxcli/usererr"
-	"go.jetpack.io/devbox/internal/nix"
+	"go.jetpack.io/devbox/internal/ux"
 )
 
 func globalCmd() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:    "global",
-		Short:  "Manage global devbox packages",
-		Hidden: false,
-	}
-
-	cmd.AddCommand(globalAddCmd())
-	cmd.AddCommand(globalInstallCmd())
-	cmd.AddCommand(globalListCmd())
-	cmd.AddCommand(globalPullCmd())
-	cmd.AddCommand(globalRemoveCmd())
-	cmd.AddCommand(globalShellenvCmd())
 
-	return cmd
-}
+	globalCmd := &cobra.Command{}
```

Comment:
`globalCmd` is used in the definition itself (see `PersistentPreRunE`)

Reply:
Oh, sorry to miss it...

Reply:
How about adding `config` as a persistent flag to global command so that its sub-commands could use it and you don't need to write in this way (pass `globalCmd` as an argument when initializing it)?

Reply:
That's a good idea, I'll try it out. I'm not sure how the persistent flag behaves if the subcommand declares the flag (if the persistent value populates the subcommand flag then this would work). The reason I tied myself in knots is because I don't want the config flag to really be visible for global (since that doesn't make a ton of sense, global should always point to the same directory).

Reply:
Unfortunately doesn't work because the subcommand flag overrides the parent command flag. Since these commands are used in global and non-global context we can't really remove the flag. One alternative I considered was passing the flag into the subcommand constructor, but this is messier and requires us to modify each subcommand.

I agree current solution is ugly, but can't think of a better one.